website/docs/en/config/experiments.mdx

```diff
@@ -276,69 +276,65 @@ export default {
 
 <ApiMeta addedVersion="1.1.0-beta.0" />
 
-<PropertyType
-  type="boolean | Incremental"
-  defaultValueList={[
-    { defaultValue: 'false', mode: 'production' },
-    { defaultValue: '{ make: true, emitAssets: true }', mode: 'development' },
-  ]}
-/>
-
-```ts
-type Incremental = {
-  make?: boolean;
-  inferAsyncModules?: boolean;
-  providedExports?: boolean;
-  dependenciesDiagnostics?: boolean;
-  sideEffects?: boolean;
-  buildChunkGraph?: boolean;
-  moduleIds?: boolean;
-  chunkIds?: boolean;
-  modulesHashes?: boolean;
-  modulesCodegen?: boolean;
-  modulesRuntimeRequirements?: boolean;
-  chunksRuntimeRequirements?: boolean;
-  chunksHashes?: boolean;
-  chunksRender?: boolean;
-  emitAssets?: boolean;
-};
-```
-
-Whether to enable incremental rebuild to speed up the rebuild speed. It is recommended to enable it only during development.
-
-```js title="rspack.config.mjs"
-const isDev = process.env.NODE_ENV === 'development';
-
-export default {
-  mode: isDev ? 'development' : 'production',
-  experiments: {
-    incremental: isDev,
-  },
-};
-```
-
-`true` means enable incremental for all stages. `false` means disable incremental for all stages. Incremental can also be enabled only for specified partial stages:
-
-```js title="rspack.config.mjs"
-export default {
-  experiments: {
-    // enable incremental for all stages
-    incremental: true,
-    // disable incremental for all stages
-    incremental: false,
-    // enable only for specified partial stages
-    incremental: {
-      make: true,
-      emitAssets: true,
-    },
-  },
-};
-```
+- **Type:** `boolean | 'none' | 'safe' | 'advance' | 'advance-silent' | Incremental`
+- **Default:** `'safe'`
+
+Whether to perform the rebuild incrementally to speed up the rebuild and HMR. Two configuration ways are provided:
+
+1. Preset: including `boolean | 'none' | 'safe' | 'advance' | 'advance-silent'`
+   - `false | 'none'`: Turn off, and it will not be enabled for any stage.
+   - `'safe'`: Enable incremental for `make` and `emitAssets` stages, This is also the current default behavior of Rspack.
+   - `true | 'advance-silent'`: Enable incremental for all stages to maximize the performance of rebuilds and HMR. After these stages become stable in the future, we will set this option as the default behavior of Rspack.
+   - `'advance'`: The same as the above, but it will throw a warning to prompt the user in cases where incremental is not friendly. This option serves as a supplement to `true | 'advance-silent'` to help users troubleshoot configurations that may potentially affect the performance of incremental rebuilds.
```

Comment:
```suggestion
   - `'advance'`: The same as above, but it will detect cases that are unfriendly to incremental build and throw warnings to users (e.g., incorrect configurations). This option can help you to identify potential issues affecting incremental build performance.
```